cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/common/TopicMinIsrCacheTest.java

```diff
@@ -0,0 +1,183 @@
+
+
+    EasyMock.replay(describeConfigsResult, describedConfigsFuture, describedConfigsFuture2, topicConfig2);
+    topicMinIsrCache.putTopicMinIsr(describeConfigsResult);
+    Map<String, TopicMinIsrCache.MinIsrWithTime> minIsrWithTimeByTopic = topicMinIsrCache.minIsrWithTimeByTopic();
+    assertEquals(2, minIsrWithTimeByTopic.size());
+    assertEquals(Integer.parseInt(MIN_IN_SYNC_REPLICAS_VALUE), minIsrWithTimeByTopic.get(MOCK_TOPIC).minISR());
+    assertEquals(Integer.parseInt(MIN_IN_SYNC_REPLICAS_VALUE_2), minIsrWithTimeByTopic.get(MOCK_TOPIC_2).minISR());
+    assertEquals(MOCK_TIME_MS, minIsrWithTimeByTopic.get(MOCK_TOPIC).timeMs());
+    assertEquals(MOCK_TIME_MS, minIsrWithTimeByTopic.get(MOCK_TOPIC_2).timeMs());
+
+    EasyMock.verify(_mockTime, describeConfigsResult, describedConfigsFuture, describedConfigsFuture2, topicConfig2);
+
+    // 3. Test put for 2 described topic configs and one described config having InterruptedException.
+    EasyMock.reset(describeConfigsResult, describedConfigsFuture);
+
+    KafkaFuture<Config> describedConfigsFuture3 = EasyMock.createMock(KafkaFuture.class);
+    Config topicConfig3 = EasyMock.createMock(Config.class);
+
+    describeConfigsWithMultipleValues = new HashMap<>(2);
+    describeConfigsWithMultipleValues.put(MOCK_TOPIC_RESOURCE_2, describedConfigsFuture);
+    describeConfigsWithMultipleValues.put(MOCK_TOPIC_RESOURCE_3, describedConfigsFuture3);
+
+    EasyMock.expect(describeConfigsResult.values()).andReturn(describeConfigsWithMultipleValues);
+    EasyMock.expect(describedConfigsFuture.get()).andThrow(new InterruptedException());
+    EasyMock.expect(describedConfigsFuture3.get()).andReturn(topicConfig3);
+    EasyMock.expect(topicConfig3.get(EasyMock.eq(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG)))
+            .andReturn(new ConfigEntry(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG, MIN_IN_SYNC_REPLICAS_VALUE_3));
+
+
+    EasyMock.replay(describeConfigsResult, describedConfigsFuture, describedConfigsFuture3, topicConfig3);
+    topicMinIsrCache.putTopicMinIsr(describeConfigsResult);
+    minIsrWithTimeByTopic = topicMinIsrCache.minIsrWithTimeByTopic();
+    assertEquals(3, minIsrWithTimeByTopic.size());
+    assertEquals(Integer.parseInt(MIN_IN_SYNC_REPLICAS_VALUE), minIsrWithTimeByTopic.get(MOCK_TOPIC).minISR());
+    assertEquals(Integer.parseInt(MIN_IN_SYNC_REPLICAS_VALUE_2), minIsrWithTimeByTopic.get(MOCK_TOPIC_2).minISR());
+    assertEquals(Integer.parseInt(MIN_IN_SYNC_REPLICAS_VALUE_3), minIsrWithTimeByTopic.get(MOCK_TOPIC_3).minISR());
+    assertEquals(MOCK_TIME_MS, minIsrWithTimeByTopic.get(MOCK_TOPIC).timeMs());
+    assertEquals(MOCK_TIME_MS, minIsrWithTimeByTopic.get(MOCK_TOPIC_2).timeMs());
+    assertEquals(MOCK_TIME_MS, minIsrWithTimeByTopic.get(MOCK_TOPIC_3).timeMs());
+
+    EasyMock.verify(_mockTime, describeConfigsResult, describedConfigsFuture, describedConfigsFuture3, topicConfig3);
+
+    // 4. Test putting one more new entry when the cache is full.
```

Comment:
Should we add one more test to make sure cache is empty after cache expire time?